lib/ansible/executor/action_write_locks.py

```diff
@@ -36,8 +37,7 @@
 
     # These plugins are called directly by action plugins (not going through
     # a strategy).  We precreate them here as an optimization
-    mods = set(p['name'] for p in Facts.PKG_MGRS)
+    mods = set(C.PKG_MGR_MODULES)
```

Comment:
The main requirement for me is leaving it in module_utils.  Why do you think that there shouldn't be a dependency from ansible into module_utils?

(Also default_squash_actions is entirely different.  default_squash_actions is only used by controller code.  PKG_MGR_MODULES is used primarily by remote-side code.)